mapflow/functional/service/data_catalog.py

```diff
@@ -167,7 +169,14 @@ def selected_mosaic(self) -> Optional[MosaicReturnSchema]:
             return None
         return first[0]
 
-    def selected_image(self, limit=1) -> Optional[ImageReturnSchema]:
-        self.view.selected_images_ids(limit=limit)
-        image = self.images[0]
-        return image
\ No newline at end of file
+    def selected_images(self, limit=None) -> List[MosaicReturnSchema]:
+        ids = self.view.selected_images_ids(limit=limit)
```

Comment:
This search by ID looks heavy. Maybe we could rely on the images order (the same in self.images and in the table) and index them just by table positions? And if not, maybe better to store them not in list, but in a dict, like mosaics.

Similarly, when we get "selected_mosaics" we don't need to iterate over ".items", as ID is the key in the dict; we should just use `selected_mosaics = [self.mosaics[id] for id in selected_mosaic_ids]`